src/framework/audio/internal/plugins/registeraudiopluginsscenario.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "registeraudiopluginsscenario.h"
+
+#include <QApplication>
+
+#include "translation.h"
+#include "log.h"
+
+using namespace mu::audio;
+using namespace mu::framework;
+
+void RegisterAudioPluginsScenario::init()
+{
+    TRACEFUNC;
+
+    m_progress.finished.onReceive(this, [this](const ProgressResult& res) {
+        if (res.ret.code() == static_cast<int>(Ret::Code::Cancel)) {
+            m_aborted = true;
+        }
+    });
+}
+
+mu::Ret RegisterAudioPluginsScenario::registerNewPlugins()
+{
+    TRACEFUNC;
+
+    io::paths_t newPluginPaths;
+
+    for (IAudioPluginsScannerPtr scanner : scannerRegister()->scanners()) {
+        io::paths_t paths = scanner->scanPlugins();
+
+        for (const io::path_t& path : paths) {
+            if (!knownPluginsRegister()->exists(path)) {
+                newPluginPaths.push_back(path);
+            }
+        }
+    }
+
+    startPluginsRegistration(newPluginPaths);
```

Comment:
I think it's better to call it `processXXX`, because `start` means it's like it's an asynchronous operation, and there will be a finish somewhere.